antiplag/views.py

```diff
@@ -83,14 +100,24 @@ def post(self, request):
                 )
 
         else:
-            text_raw = request.body.decode()
+            data = json.loads(request.body)
+            text_raw = data["text"]
+            email = data["email"] if "email" in data else None
 
             if not text_raw.strip():
                 return Response(
                     {"error": _("No text was specified.")},
                     status=status.HTTP_400_BAD_REQUEST,
                 )
 
+            if email is not None and email_valid(email) is False:
+                return Response(
+                    {"error": _("Unrecognized email address format.")},
+                    status=status.HTTP_400_BAD_REQUEST,
+                )
+            submission.email = email
+            submission.save()
+
```

Comment:
You dont need to do the same logic twice. Just validate it at the beginning.

Reply:
Getting the email depends on the request header. Then you'd check the header twice.

Reply:
@OliverKanat No we can't, because there is a difference in the header we get based on if it's files or text. Files is multiform and text is now json. Because Post.get() would crash on the application/json header we isolated this. It was like that before this change. @MCFreddie777  Any further comments on this?

Reply:
OK - I see. The issue is somewhere else - but that's another topic. Keep it duplicated.